state/state.go

```diff
@@ -102,11 +102,12 @@ func (s State) GetValidators() (last *types.ValidatorSet, current *types.Validat
 // Create a block from the latest state
 
 // MakeBlock builds a block with the given txs and commit from the current state.
-func (s State) MakeBlock(height int64, txs []types.Tx, commit *types.Commit) (*types.Block, *types.PartSet) {
-	// build base block
+func (s State) MakeBlock(height int64, txs []types.Tx, commit *types.Commit, blockTimeIota int) (*types.Block, *types.PartSet) {
```

Comment:
This used to be a gross function with many many args. I had hoped that MakeBlock could more directly reflect the block structure - header, txs, commit. By moving it to be a method on the State, we achieved that. Ideally, we wouldn't add more args, and everything else would come in through the state. We did add Evidence to the block, but we just use the block.AddEvidence method for it. So ideally, we wouldn't have to pass blockTimeIota here.

Actually, I don't think blockTimeIota belong in ConsensusConfig - it is more like ConsensusParams! ConsensusConfig is not required to validate blocks, and nodes can tweak them subjectively. Not so for the ConsensusParams. Unlike the other timeout parameters, blockTimeIota is actually used in validating a block in the state package, so it can't be tweaked.

So I think we need to move blockTimeIota into ConsensusParams. That should also eliminate the need to pass it in to this method, since we have State.ConsensusParams. 

As for the other blockTime params, my understanding is they do belong in ConsensusConfig.

Reply:
ConsensusConfig.Timeout*, MaxBlockSize* are maybe others are on the same level of configuration as blockTimeIota.  It doesn't make sense to separate them, they're intertwined.  We can make them all consensus params, that would be OK.

Reply:
But it would suck if people's apps messed up the params to the point that it breaks consensus.  So maybe we should preserve a way to override these configs if they are present as overrides in consensus config.